app/models/school.rb

```diff
@@ -119,15 +129,11 @@ def eligible_for_student_loans_as_current_school?
   end
 
   def eligible_for_maths_and_physics?
-    MathsAndPhysics::SchoolEligibility.new(self).check
+    MathsAndPhysics::SchoolEligibility.new(self).eligible_current_school?
   end
 
   def state_funded?
-    STATE_FUNDED_SCHOOL_TYPE_GROUPS.include?(school_type_group) && school_type != "other_independent_special_school"
-  end
-
-  def special?
-    SPECIAL_SCHOOL_TYPES.include?(school_type)
+    (STATE_FUNDED_SCHOOL_TYPE_GROUPS.include?(school_type_group) && school_type != "other_independent_special_school") || secure_unit?
```

Comment:
I don't think this is the right place to include secure units. `state_funded?` should always return true of false based on the funding type of the school. If we want to include non-state funded secure units in the list of eligible schools then we should have that logic in the eligibility check, not here as the exception is for eligibility, not for funding.